From fedce2012c30e90fb5498ecfc65a28f0fb14a3a8 Mon Sep 17 00:00:00 2001
From: Lim Siew Hoon <siew.hoon.lim@intel.com>
Date: Tue, 26 Mar 2024 09:02:15 +0800
Subject: [PATCH] [VP] Fix 4K monitor failed to run videowall 4k on multi
 channel

Enlarge max patched/kernel/csc matrix number to fix 4k multi channel issue.

[Internal]
Commit_Type: Bugfix
Platforms: Xe_Lpm
OS: Linux
Feature impact: CSC_Scal_Rot_Mirr
Resolves: VSMGWL-72000
Related-to: VSMGWL-72000
Klocwork: PASS
TP_Passed: N/A
IP Scan: PASS

Backport from internal PR code fixed:
https://github.com/intel-innersource/drivers.gpu.unified/pull/158683

Signed-off-by: Lim, Siew Hoon <siew.hoon.lim@intel.com>
---
 media_common/agnostic/common/vp/kdll/hal_kerneldll_next.h | 8 ++++----
 .../agnostic/common/vp/kdll/hal_kerneldll_next.c          | 3 ++-
 2 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/media_common/agnostic/common/vp/kdll/hal_kerneldll_next.h b/media_common/agnostic/common/vp/kdll/hal_kerneldll_next.h
index 8e2ab371e..f68040923 100644
--- a/media_common/agnostic/common/vp/kdll/hal_kerneldll_next.h
+++ b/media_common/agnostic/common/vp/kdll/hal_kerneldll_next.h
@@ -54,17 +54,17 @@
 
 #define DL_MAX_SEARCH_FILTER_SIZE 10  // max number of entries to describe a compositing filter
 
-#define DL_MAX_KERNELS 150         // Max component kernels to combine
+#define DL_MAX_KERNELS 256         // Max component kernels to combine
 #define DL_MAX_PATCH_DATA_SIZE 64  // Max size of a patch block
 #define DL_MAX_PATCH_BLOCKS 8      // Max number of blocks to patch per patch data
-#define DL_MAX_PATCHES 4           // Max patches to use
+#define DL_MAX_PATCHES 8           // Max patches to use
 #define DL_MAX_EXPORT_COUNT 64     // size of the symbol export table
 
 #define DL_MAX_COMBINED_KERNELS 64       // Max number of kernels in cache
 #define DL_MAX_SYMBOLS 100               // max number of import/export symbols in a combined kernels
 #define DL_MAX_KERNEL_SIZE (140 * 1024)  // max output kernel size
 
-#define DL_CSC_MAX 6                      // 6 CSC matrices max
+#define DL_CSC_MAX 8                      // 8 CSC matrices max
 #define DL_MAX_SEARCH_NODES_PER_KERNEL 6  // max number of search nodes for a component kernel (max tree depth)
 #define DL_MAX_COMPONENT_KERNELS 25       // max number of component kernels that can be combined
 
@@ -984,4 +984,4 @@ bool KernelDll_SetupFunctionPointers_Ext(
 }
 #endif
 
-#endif  // __HAL_KERNELDLL_NEXT_H__
\ No newline at end of file
+#endif  // __HAL_KERNELDLL_NEXT_H__
diff --git a/media_softlet/agnostic/common/vp/kdll/hal_kerneldll_next.c b/media_softlet/agnostic/common/vp/kdll/hal_kerneldll_next.c
index 151f3b039..68ddea2bb 100644
--- a/media_softlet/agnostic/common/vp/kdll/hal_kerneldll_next.c
+++ b/media_softlet/agnostic/common/vp/kdll/hal_kerneldll_next.c
@@ -2594,6 +2594,7 @@ bool KernelDll_SetupCSC(
                 // Exceeded number of CSC matrices allowed
                 if (matrix_count == DL_CSC_MAX)
                 {
+                    VP_RENDER_ASSERTMESSAGE("Exceeded number of CSC matrices allowed!");
                     return false;
                 }
 
@@ -4466,4 +4467,4 @@ bool KernelDll_SetupFunctionPointers_Ext(
 
 #ifdef __cplusplus
 }
-#endif  // __cplusplus
\ No newline at end of file
+#endif  // __cplusplus
-- 
2.40.1

